### PR TITLE
Kinetis usb fix

### PR DESCRIPTION
**Pull request by c1728p9**
When a SETUP packet comes immediately after a control transfer with a data in stage, there is a chance that the SETUP packet will be dropped. This is because no buffers are available to recieve the setup packet. This patch turns on double buffering to prevent SETUP packets from being dropped.

### CODE REVIEW

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

File: source/hic_hal/freescale/k20dx/usbd_MK20D5.c [hunk 1]
```diff
@@ -526,7 +585,51 @@ U32 USBD_GetError(void)
  */
 void USB0_IRQHandler(void)
 {
-    NVIC_DisableIRQ(USB0_IRQn);
+    uint32_t istat, num, dir, ev_odd;;
```
Comment (sg-) [inline source/hic_hal/freescale/k20dx/usbd_MK20D5.c]: double ;;

File: source/hic_hal/freescale/kl26z/usbd_MKL26Z4.c [hunk 2]
```diff
@@ -529,20 +585,72 @@ U32 USBD_GetError(void)
  */
 void USB0_IRQHandler(void)
 {
-    NVIC_DisableIRQ(USB0_IRQn);
+    uint32_t istat, num, dir, ev_odd;;
```
Comment (sg-) [inline source/hic_hal/freescale/kl26z/usbd_MKL26Z4.c]: double ;;

Comment (c1728p9) [inline source/hic_hal/freescale/kl26z/usbd_MKL26Z4.c]: Good catch. Fixed.

**Comment by c1728p9** [comment]
@sg- the only difference is the header inclusion - "MKL26Z4.h" vs "MK20D5.h". Having one file would make maintenance a lot easier. Does creating a common kinesis directory sound good?

**Comment by sg-** [comment]
yep!

**Comment by flit** [comment]
The `fsl_device_registers.h` header is intended to be the common include for situations like this.

**Comment by c1728p9** [comment]
Thanks for the suggestion @flit, "fsl_device_registers.h" allows the same header to be used for both. PR updated to use a single USB driver file.

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by sg-** [comment]
LGTM was the exclusive load and store from protected xor and and removed a while ago? That is the only patch I recall seeing made to the k20 driver years ago

**Comment by c1728p9** [comment]
protected_and, protected_or, and protected_xor still exist and are being used. Since the driver is single threaded it doesn't matter as much now though.

**Pull request merged**